BaseLib/LogogCustomCout.h

```diff
@@ -48,12 +54,15 @@ class LogogCustomCout : public logog::Target
 
 	virtual int Output( const LOGOG_STRING &data )
 	{
+#if defined(USE_MPI) || defined(USE_PETSC)
+		if (_is_rank0)
```

Comment:
If you include "_is_rank0" in all cases, not only for MPI and PETSC, and set it to true in the other cases, then the ifdefs can be omitted.